pkg/sql/logictest/testdata/logic_test/distsql_join

```diff
@@ -67,6 +67,52 @@ EXPLAIN (VERBOSE) (SELECT * FROM (SELECT a,b FROM data) NATURAL JOIN (SELECT a,b
 3  ·       table           data@primary       ·                                                                                   ·
 3  ·       spans           ALL                ·                                                                                   ·
 
+# Test planning with USING plus an extra ON condition (#20569), possibly adding another equality column.
+query T
+SELECT "URL" FROM [EXPLAIN (DISTSQL) SELECT * FROM data AS l JOIN data as r USING(a, b) WHERE l.c = r.c AND a + b + l.c + l.d + r.d < 8]
+----
+https://cockroachdb.github.io/distsqlplan/decode.html?eJzsllGL2kwUhu-_XxHO1YrzQWYSXQ0UptAbC3XLsndtLrLmVANuRiYT6LL430uSLa4xzjGMBQW9CKjzzLyceQLvG-QqxXnyggVEP4ADAwEMAmAQAoMRxAw2Wi2wKJSuljTALP0Nkc8gyzelqX6OGSyURojewGRmjRDBU_K8xkdMUtTAIEWTZOv6kI3OXhL9KtPEJMDgoTSRJzmTgsmAyRDiLQNVmveddxs-v3qrpFjtb_YXhHgbMyhMskSI-JZdRU5xNOduN6VT1Ji2dxsyKYZMBsMqRq_VHSP4hnqJX1WWt0ewxl_m7gM--KSz5Wr_p2o0c--u-kjuDT0pBtUzqJ9h_ZwMvJ-l7wcLbxJF0Wz-NPA-z7-8M6NqxfgEpuMOmLxnctK6id18gxPmW-ZdM-sc01z9rzatZd0Hh3sH84sVkF9JTnE05-1FOf-LIi7WA34lOcXRnDdfz-9rcLEe8CvJKY7mvPl6fl_Di_WAX0lOcTTnzdd_W5w7hvCIxUblBZ7UjP1qjJgusbmsQpV6gd-1WtTHNF8faq7ueSkWpvlXNF9mefNXFfB0eOwCT11g7pSbj-w07zEy0Q8eu8BTF5g75W6N7IAWbdr_SAf2eQdWmO_PzG_ToYvgdpgQ3A4TgtthSnCCJgQfuQhuhwnB7TAhuB2mBCdoQvCxi-D3LoraYUJRO0woaocpRQmaUHTioqgdJhS1w4SidphSlKAJRacuinKnnkDQhKQETVhK0JSmFE51Bbey4NYW3OqCY19wKwzcqTHwg8rQy1Y7Tdlqpylb7TRpK4FTtvYpS4d31qct9aUpW3v1pd44ZetBebDaGm__-xMAAP__1afPNg==
+
+query IIIIII rowsort
+SELECT * FROM data AS l JOIN data as r USING(a, b) WHERE l.c = r.c AND a + b + l.c + l.d + r.d < 8
+----
+1  1  1  1  1  1
```

Comment:
nit: it might be best to add `colnames` too to make this easier to read (and to verify where the merged columns are) in the future.